### DISCUSSION TITLE
Freeze fs before snapshot

### DISCUSSION THREAD

**Discussion by danbryan** [backube/snapscheduler#205] [Q&A] [answered]
Does anyone know if snapscheduler will freeze the PVC from being written to by the pods, while the snapshot is being created?

**Comment by JohnStrunk** [comment, marked as answer]
SnapScheduler doesn't take any specific action. I'm not entirely sure what level of coordination you're asking about specifically.

The CSI driver for your storage system is responsible for the actual creation of the snapshot. This may entail freezing the fs so that it can get a crash-consistent image. Whether or not it performs an actual "freeze", it is responsible for obtaining a crash-consistent image of the data.

The other level of coordination would be with the application, ensuring that the application has flushed any data to quiesce its state. This level of coordination isn't available via a standard interface in Kubernetes (today). See kubernetes/enhancements#1995 for the current proposal. While there are some proprietary solutions, it's not something that can be done in a general way today.

The short answer is that you should end up w/ a crash-consistent image of your data. The specifics of how that is accomplished is the responsibility of the CSI driver. SnapScheduler just automates creating VolumeSnapshot objects based on a schedule.
